Content.Shared/SS220/LyingDownOnBuckledEntity/Systems/LyingDownOnBuckledEntitySystem.cs

```diff
@@ -0,0 +1,322 @@
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, EndCollideEvent>(OnEndCollide);
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, DamageChangedEvent>(OnDamageChanged);
+
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, UpdateCanMoveEvent>(OnCanMoveUpdate);
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, GetVerbsEvent<AlternativeVerb>>(OnAlternativeVerb);
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, RaisePetDoAfterEvent>(OnRaiseDoAfter);
+        SubscribeLocalEvent<LyingDownOnBuckledEntityComponent, LieDownDoAfterEvent>(OnLieDownDoAfter);
+
+        SubscribeLocalEvent<LyingPetComponent, UnbuckleAttemptEvent>(OnUnbuckleAttempt);
+        SubscribeLocalEvent<LyingPetComponent, UnstrapAttemptEvent>(OnUnstrapAttempt);
+        SubscribeLocalEvent<HumanoidAppearanceComponent, GetVerbsEvent<AlternativeVerb>>(OnHumanoidAlternativeVerb);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<LyingPetComponent>();
+        while (query.MoveNext(out var uid, out var lyingPet))
+        {
+            if (_timing.CurTime < lyingPet.NextSecond)
+                continue;
+
+            lyingPet.NextSecond = _timing.CurTime + TimeSpan.FromSeconds(1);
+
+            if (lyingPet.Damage != null && !_mobState.IsDead(uid))
+                _damageable.TryChangeDamage(uid, lyingPet.Damage, interruptsDoAfters: false);
+        }
+    }
+
+    private void OnShutdown(Entity<LyingDownOnBuckledEntityComponent> entity, ref ComponentShutdown args)
+    {
+        if (entity.Comp.IsLying)
+            StandUp(entity);
+
+        Dirty(entity);
+    }
+
+    private void OnStandUpAction(Entity<LyingDownOnBuckledEntityComponent> entity, ref StandUpActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        StandUp(entity);
+        args.Handled = true;
+    }
+
+    private void OnMobStateChanged(Entity<LyingDownOnBuckledEntityComponent> entity, ref MobStateChangedEvent args)
+    {
+        if (entity.Comp.IsLying &&
+            args.NewMobState is not MobState.Alive)
+            StandUp(entity);
+    }
+
+    private void OnDamageChanged(Entity<LyingDownOnBuckledEntityComponent> entity, ref DamageChangedEvent args)
+    {
+        if (args.DamageDelta is { } damageDelta &&
+            damageDelta.GetTotal() >= entity.Comp.DamagetThreshold)
+            StandUp(entity);
+    }
+
+    private void OnEndCollide(Entity<LyingDownOnBuckledEntityComponent> entity, ref EndCollideEvent args)
+    {
+        if (entity.Comp.LyingOn == args.OtherEntity)
+            StandUp(entity);
+    }
+    private void OnCanMoveUpdate(Entity<LyingDownOnBuckledEntityComponent> entity, ref UpdateCanMoveEvent args)
```

Comment:
Заодно пустую строку закинь тут

```suggestion
    }
    
    private void OnCanMoveUpdate(Entity<LyingDownOnBuckledEntityComponent> entity, ref UpdateCanMoveEvent args)
```